components/region_cache_memory_engine/src/range_manager.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright 2024 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use engine_traits::CacheRange;
+
+use crate::engine::{RagneCacheSnapshotMeta, SnapshotList};
+
+#[derive(Debug, Default)]
+pub struct RangeMeta {
+    id: u64,
+    range_snapshot_list: SnapshotList,
+    evicted: bool,
```

Comment:
Is `evicted` needed after all? When a range is evicted, it can be moved to historical_ranges.

Reply:
yes, no needed.